handlers.go

```diff
@@ -74,6 +75,22 @@ func logSigningRequestFailure(sigreq formats.SignatureRequest, sigresp formats.S
 	}).Info(fmt.Sprintf("signing operation failed with error: %v", err))
 }
 
+// If the x5u is a local file, mirror it via the `x5u` endpoint instead.
+func mirrorLocalX5U(r *http.Request, response formats.SignatureResponse) string {
+	parsedX5U, err := url.Parse(response.X5U)
```

Comment:
Now that there's less to look at I'm wondering: could we do this URL rewriting earlier in the process? That is to say, is it possible to rewrite the X5U URL on the signer at signer creation time and not need all this happening at request time? 

A couple follow-up questions I have after that: Would doing that rewrite earlier (if we can) allow us to simplify `contentsignaturepki.GetX5U`? And would doing the url rewrite earlier also mean the `getMirroredX5U` helper could go away since it doesn't need to know about files anymore? (I know I have a comment on `getMirroredX5U` wondering about whether it needs to exist given the current state of the code, but I'm asking here in case that other question gets a "yes it needs to exist" answer)

Reply:
In theory it can be done statically by changing the X5U in the signer config to match what gets served by `http.FileServer` but to figure that out what the X5U should be, autograph would need to know its hostname to fully form the URL. And the hostname might not even be knowable at startup if it's running behind proxies or across container boundaries.

On the other hand, handling this at the time of the request is easy because we can just copy the hostname from the HTTP request headers.